DentOS_Framework/DentOsTestbedLib/src/dent_os_testbed/lib/traffic/ixnetwork/ixnetwork_ixia_client_impl.py

```diff
@@ -277,162 +293,131 @@ def __update_field(self, field, value):
         field.Auto = False
         field.update(**param)
 
-    def set_l4_traffic(self, config_element, ipv4_stack, pkt_data):
-        if "ipproto" not in pkt_data:
-            return
-        if pkt_data["ipproto"] not in ["tcp", "udp", "icmpv1", "icmpv2"]:
-            return
-        ipproto_template = IxnetworkIxiaClientImpl.ixnet.Traffic.ProtocolTemplate.find(
-            StackTypeId="^{}$".format(pkt_data["ipproto"])
-        )
-        l4_stack = config_element.Stack.read(ipv4_stack.AppendProtocol(ipproto_template))
-        if "dstPort" in pkt_data:
-            if ":" in pkt_data["dstPort"]:
-                start, step, count = pkt_data["dstPort"].split(":")
-                value = {"type": "increment", "start": start, "step": step, "count": count}
-            else:
-                value = pkt_data["dstPort"]
-            self.__update_field(l4_stack.Field.find(FieldTypeId=f"{pkt_data['ipproto']}.header.dstPort"),
-                                value)
-        if "srcPort" in pkt_data:
-            self.__update_field(l4_stack.Field.find(FieldTypeId=f"{pkt_data['ipproto']}.header.srcPort"),
-                                pkt_data["srcPort"])
-
-        if "icmpType" in pkt_data:
-            self.__update_field(l4_stack.Field.find(FieldTypeId=f"{pkt_data['ipproto']}.message.messageType"),
-                                pkt_data["icmpType"])
-        if "icmpCode" in pkt_data:
-            self.__update_field(l4_stack.Field.find(FieldTypeId=f"{pkt_data['ipproto']}.message.codeValue"),
-                                pkt_data["icmpCode"])
-
-    def set_ethernet_traffic(self, device, name, pkt_data, traffic_type):
-        # create an ipv4 traffic item
-        ipv4_template = IxnetworkIxiaClientImpl.ixnet.Traffic.ProtocolTemplate.find(
-            StackTypeId="^ipv4$"
-        )
-        vlan_template = IxnetworkIxiaClientImpl.ixnet.Traffic.ProtocolTemplate.find(
-            StackTypeId="^vlan$"
-        )
-        for ip1, ep1, rep1 in zip(
-            IxnetworkIxiaClientImpl.ip_eps,
-            IxnetworkIxiaClientImpl.eth_eps,
-            IxnetworkIxiaClientImpl.raw_eps,
-        ):
-            if "ip_source" in pkt_data and ip1.Name not in pkt_data["ip_source"]:
-                continue
-            if "ep_source" in pkt_data and ep1.Name not in pkt_data["ep_source"]:
+    @classmethod
+    def __create_traffic_items(cls, device, pkt_data, name):
+        traffic_type = pkt_data.get("type", "ipv4")
```

Comment:
Can you use the same attribute name - `version` instead of `type`?

Reply:
Not in this case, no. This variable is not the IP version (ipv4/ipv6), but it is a traffic type (raw/ethernet/ipv4/ipv6). We are using this naming convention in all of our tests, so changing this is not an option.
```python
streams = {
    "traffic": {
        "type": "ipv4",  # <--- could be 'raw', 'ethernet', 'ipv4', 'ipv6'
        "rate": 1000,
        "ip_source": src,
        "ip_destination": dst,
    }
}

await tgen_utils_setup_streams(tgen_dev, None, streams)
```